### PR TITLE
Attempt to fix PPC/PPC64 ABI issues

### PR DESCRIPTION
**Pull request by kinke**
Continuation of https://github.com/ldc-developers/ldc/pull/1683.

### CODE REVIEW

**Comment by kinke** [comment]
Kai mentioned that LDC was able to compile itself back then, but that one test still failed. I assume it's the one wrt. reversed sorting, which was also a problem for ARM until the arguments order has been reversed for `extern(D)`.

**Comment by kinke** [comment]
Note that https://github.com/ldc-developers/ldc/pull/1680 was merged, i.e., Kai's fixes are in ltsmaster, but not in master.

File: gen/abi-mips64.cpp [hunk 1]
```diff
@@ -44,6 +48,10 @@ struct MIPS64TargetABI : TargetABI {
   }
 
   void rewriteFunctionType(TypeFunction *tf, IrFuncTy &fty) override {
+    if (!fty.ret->byref) {
+      rewriteArgument(fty, *fty.ret);
+    }
```
Comment (kinke) [inline gen/abi-mips64.cpp]: Note that `MIPS64TargetABI::rewriteArgument()` is empty and needs to be implemented, so this ABI has probably never been properly tested. Almost all ABIs rewrite a non-sret return value just like a regular argument of the same type.

**Comment by kinke** [comment]
@joakim-noah, @smolt: It would be nice if you could verify everything still works.

**Comment by joakim-noah** [comment]
Compiling ltsmaster natively on Android/ARM with the ARM portion of this PR, then will try the full PR with master.  That'll cover ARM soft-float, somebody with an armhf board will have to test that separately.

**Comment by kalev** [comment]
I tested the pull request and it fixes the ppc64 issue, https://github.com/ldc-developers/ldc/issues/1904 for me. Thanks!

I ran it through Fedora's armv7hl, x86_64, ppc64le, ppc64, i686 builders and ldc can successfully build itself on all of these now.

**Comment by kinke** [comment]
Excellent, thank you very much!

**Comment by joakim-noah** [comment]
Full stdlib and dmd testsuite passes with this PR on Android with ltsmaster and master.

**Comment by kinke** [comment]
See https://github.com/ldc-developers/ldc/pull/865 and https://github.com/dlang/dmd/pull/5232 as to why druntime and the compiler itself currently depend on the reversed order.

**Pull request merged**